libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3696,20 +3729,28 @@ instance BaseUniverse fn => Functions (SetFn fn) fn where
                       , forAll disjoint $ \e -> e `satisfies` es
                       , assert $ lit (must Set.\\ s) `subset_` disjoint
                       ]
-            MemberSpec [e]
-              | s `Set.isSubsetOf` e -> typeSpec (SetSpec (Set.difference e s) (MemberSpec $ Set.toList e) mempty)
+            MemberSpec es
+              | NE.length es == 1
+              , Just ys <- NE.nonEmpty (Set.toList (NE.head es))
+              , s `Set.isSubsetOf` (NE.head es) ->
+                  typeSpec
+                    ( SetSpec
+                        (Set.difference (NE.head es) s)
+                        (MemberSpec ys)
+                        mempty
+                    )
               -- TODO: improve this error message
               | otherwise ->
                   ErrorSpec (pure "propagateSpecFun Union MemberSpec singleton with bad literal")
-            -- This risks blowing up too much, don't build sets of sets
-            MemberSpec {} -> ErrorSpec (pure "propagateSpecFun Union MemberSpec")
+    -- This risks blowing up too much, don't build sets of sets
+
```

Comment:
This results in a weird error message that doesn't correspond to reality when you have more than one thing in the member spec now